File: 01_Language/01_Functions/go/src/functions/addslashes.go
package functions

import "strings"

func Addslashes(str string) string {
	str = strings.ReplaceAll(str, "\\", "\\\\")
	str = strings.ReplaceAll(str, "'", "\\'")
	str = strings.ReplaceAll(str, "\"", "\\\"")
	str = strings.ReplaceAll(str, "\x00", "\\0")
	return str
}

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/AddSlashes.java
package com.onevgo.functions;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class AddSlashes {
    private static final Pattern PATTERN = Pattern.compile("[\\\\'\"\u0000]");

    public static String addSlashes(String s) {
        Matcher matcher = PATTERN.matcher(s);
        boolean result = matcher.find();
        if (result) {
            StringBuffer sb = new StringBuffer();
            do {
                matcher.appendReplacement(sb, "\u0000".equals(matcher.group(0)) ? "\\\\0" : "\\\\$0");
                result = matcher.find();
            } while (result);
            matcher.appendTail(sb);
            return sb.toString();
        }
        return s;
    }

    public static void main(String[] args) {
        String str = "' \" \\ \0";
        System.out.println(str);
        System.out.println(addSlashes(str));
    }
}
